kuma/javascript/src/index.jsx

```diff
@@ -15,10 +16,11 @@ if (container) {
     // be garbage collected once it is no longer in use.
     window._react_data = null; // eslint-disable-line camelcase
 
+    // Store the string catalog so that l10n.gettext() can do translations
+    localize(data.requestData.locale, data.localizationData);
```

Comment:
It seems like you could add `getLocale` to the import from `./l10n.js` above, and use `getLocale()` instead of `data.requestData.locale` in the call to `localize`. I would vote for doing that and getting rid of `requestData` altogether, since the `locale` is baked into the localization data (see comment below as well).

Reply:
I can't call getLocale() here, because this is the call that is initializing that data. getLocale() won't work until localize() has been called.

And data.localizationData does not define the locale that it is for. I could modify the backend to add a locale property to the localizationData dict, but then it wouldn't just be a copy of the react.json file...